python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -152,53 +155,107 @@ cdef Scalar _new_scalar(unique_ptr[scalar] c_obj, DataType dtype):
 
 
 @singledispatch
-def _from_py(py_val):
+def _from_py(py_val, dtype: DataType | None):
     raise TypeError(f"{type(py_val).__name__} cannot be converted to pylibcudf.Scalar")
 
 
 @_from_py.register(dict)
 @_from_py.register(list)
 @_from_py.register(datetime.datetime)
 @_from_py.register(datetime.timedelta)
-def _(py_val):
+def _(py_val, dtype: DataType | None):
     raise NotImplementedError(
         f"Conversion from {type(py_val).__name__} is currently not supported."
     )
 
 
 @_from_py.register(float)
-def _(py_val):
-    cdef DataType dtype = DataType(type_id.FLOAT64)
-    cdef unique_ptr[scalar] c_obj = make_numeric_scalar(dtype.c_obj)
-    (<numeric_scalar[double]*>c_obj.get()).set_value(py_val)
-    cdef Scalar slr = _new_scalar(move(c_obj), dtype)
-    return slr
+def _(py_val: float, dtype: DataType | None):
+    if dtype is None:
```

Comment:
Nit: Would be nice if we could 

```python
cdef DataType c_dtype
if dtype is None:
    c_dtype = DataType(type_id.FLOAT64)
else:
    c_dtype = <DataType>dtype
```
(I forgot if we need the cast in the else case)